predicate/solver/ast.py

```diff
@@ -439,7 +442,26 @@ def walk(self, fn):
         fn(self)
 
     def __str__(self):
-        return "duration({})".format(self.name)
+        return "ltduration({})".format(self.name)
+
+    def __lt__(self, val):
```

Comment:
You can make it a type check: `def __lt__(self, val: DurationLiteral)` so the parser will implement this check for you

Reply:
Thanks. Done in e7eb4c99f8ea99bad53470bca249bafa79444d38.